src/no-dead-relative-link.js

```diff
@@ -39,24 +38,54 @@ async function validateLinkNode(linkNode, context, options) {
 }
 
 async function validateRelativeLink(linkNode, context, options) {
-    let linkAbsoutePath = path.resolve(path.dirname(context.getFilePath()), linkNode.url);
-    let linkURL = new URL("file://" + linkAbsoutePath);
+    let linkAbsolutePath = path.resolve(path.dirname(context.getFilePath()), linkNode.url);
+    let linkURL = new URL("file://" + linkAbsolutePath);
     let linkedFileExtension = path.extname(linkURL.pathname);
 
-
     if (linkedFileExtension !== ".md" && options["resolve-as-markdown"] && options["resolve-as-markdown"].includes(linkedFileExtension)) {
         linkURL.pathname = linkURL.pathname.replace(linkedFileExtension, ".md");
     }
-
     if (!await fileExists(url.fileURLToPath(linkURL))) {
-        reportError(linkNode, context, `${path.basename(linkURL.pathname)} does not exist`);
+        let mappingExists = false;
+        if (options["link-route-map"]) {
+            mappingExists = await routedPathExists(context, options, linkNode);
+        }
+        if (!mappingExists) {
+            reportError(linkNode, context, `${path.basename(linkURL.pathname)} does not exist`);
+        }
         return;
-    } 
+    }
     if(linkURL.hash && path.extname(linkURL.pathname) === ".md") {
         return validateAnchorLink(url.fileURLToPath(linkURL), linkURL.hash.slice(1), linkNode, context);
     }
 }
 
+async function routedPathExists(context, options, linkNode) {
+    let linkRouteMaps = options["link-route-map"];
+    let nodeUrl = linkNode.url;
+
+    for (const mapping of linkRouteMaps) {
+        let sourceRegex = new RegExp(mapping["source"], 'g');
+        let mappedDestination = mapping["destination"];
```

Comment:
Is it intentional that the source accepts a regex but destintation does not? Or am I mis-reading this?

Reply:
Yup, if the destination is a regex as well, then the extra escape characters and slashes will be included when the text is replaced.

for example if this is the destination `../../website/static/javadocs/` -> `/..\/..\/website\/static\/javadocs\/`

Reply:
Not sure I understood that example. But, I think the usage of this option needs to be properly explained in the readme.

Reply:
If you replace the link with a regex, it will be replaced with `/..\/..\/website\/static\/javadocs\/`. You want it to be replaced with `../../website/static/javadocs/`. The replace function treats the regex as a string when it does the replacing so the escape characters and extra slashes are included. We do not want those extra characters because it will make the link invalid.

Reply:
Let's close this thread since the above code is now outdated.